docs/source/testing/mocking-responses.mdx

```diff
@@ -0,0 +1,181 @@
+---
+title: Mocking responses
+---
+
+Apollo Kotlin provides two tools to mock responses in tests:
+
+- `TestNetworkTransport`: allows to specify the GraphQL responses returned by `ApolloClient`
+- `MockServer`: a low-level HTTP server that can be used to mock any responses
+
+In most cases, `TestNetworkTransport` will be more straightforward to use in your tests, while `MockServer` can be
+useful to test specific server behavior, such as error cases, custom headers, timeouts, etc.
+
+## `TestNetworkTransport`
+
+This is a high level test API which allows to specify the responses that will be returned by your `ApolloClient`.
+
+Add the dependency to your project's `build.gradle` file:
+
+```kotlin:title=build.gradle[.kts]
+dependencies {
+  testImplementation("com.apollographql.apollo3:apollo-testing-support:x.y.z")
+}
+```
+
+Enable the `TestNetworkTransport` by passing it to the `ApolloClient` builder:
+
+```kotlin{2}
+val apolloClient = ApolloClient.Builder()
+    .networkTransport(TestNetworkTransport())
+    .build()
+```
+
+Either keep a reference to the `TestNetworkTransport` instance, or use the `testNetworkTransport` extension function to
+retrieve it:
```

Comment:
I'd recommend using the extension function like for `ApolloClient.apolloStore`. From my experience this is what most users do.

```suggestion
Use the `testNetworkTransport` extension function to retrieve the `TestNetworkTransport`:
```